smart-server/src/main/java/org/smartdata/server/command/CommandExecutor.java

```diff
@@ -371,22 +374,23 @@ private synchronized Command schedule() throws IOException {
   }
 
   private Command getCommandFromCmdInfo(CommandInfo cmdinfo) {
-    ActionBase[] actions = new ActionBase[10];
+    Action[] actions = new Action[10];
     Map<String, String> jsonParameters =
         JsonUtil.toStringStringMap(cmdinfo.getParameters());
-    String[] args = {jsonParameters.get("_FILE_PATH_")};
+    String[] args = {jsonParameters.get("_FILE_PATH_"), jsonParameters.get("_STORAGE_POLICY_")};
     // New action
-    String storagePolicy = jsonParameters.get("_STORAGE_POLICY_");
-    ActionBase current;
+    Action current;
     if (cmdinfo.getActionType().getValue() == ActionType.CacheFile.getValue()) {
-      current = new MoveToCache(ssm.getDFSClient(), ssm.getConf());
+      current = ActionRegister.getInstance().newActionFromName("CacheFile");
```

Comment:
I think it's better to have private member variable for  ActionRegister instead of using a static variable.